docs/utilities/data_classes.md

```diff
@@ -441,6 +442,41 @@ In this example, we also use the new Logger `correlation_id` and built-in `corre
     }
     ```
 
+### CloudWatch Dashboard Custom Widget
+
+=== "app.py"
+
+    ```python
+    from aws_lambda_powertools.utilities.data_classes import event_source, CloudWatchDashboardCustomWidgetEvent
+    from aws_lambda_powertools.utilities.data_classes.cloud_watch_logs_event import CloudWatchLogsDecodedData
+
+    const DOCS = `
+    ## Echo
+    A simple echo script. Anything passed in \`\`\`echo\`\`\` parameter is returned as the content of custom widget.
+
+    ### Widget parameters
+    Param | Description
+    ---|---
+    **echo** | The content to echo back
+
+    ### Example parameters
+    \`\`\` yaml
+    echo: <h1>Hello world</h1>
+    \`\`\`
+    `
+
+    @event_source(data_class=CloudWatchDashboardCustomWidgetEvent)
+    def lambda_handler(event: CloudWatchDashboardCustomWidgetEvent, context):
+
+        if event.describe:
+            return DOCS
+
+        # alternatively you can also do
+        # event.widget_context.params["echo"]
+        # event.raw_event["echo"]
+        return event["echo"]
```

Comment:
I haven't used this feature yet, but from the docs it seems that we could improve the docs by talking about the different return options:

```suggestion
        # You can directly return HTML or JSON content
        # Alternatively, you can return markdown that will be rendered by CloudWatch
        echo = event.widget_context.params["echo"]
        return { "markdown": f"# {echo}" }
```

Does it make sense?